spark/src/main/scala/eval/directive/RDDOpDirectives.scala

```diff
@@ -1,118 +1,309 @@
+      reduce({_.localMax(_)}, {(i,r) => r.localMax(i)}, {_.localMax(_)}, {(d,r) => r.localMax(d)}, {_.localMax(_)}, res1, res2)
+    }
+
+    Valid(results)
+  }
+
+  val min = Directive { case (Min(_), childResults) =>
+    val results: Result = childResults.reduce { (res1, res2) =>
+      reduce({_.localMin(_)}, {(i,r) => r.localMin(i)}, {_.localMin(_)}, {(d,r) => r.localMin(d)}, {_.localMin(_)}, res1, res2)
+    }
+
+    Valid(results)
+  }
+
+  val or = Directive { case (Or(_), childResults) =>
+    val results: Result = childResults.reduce { (res1, res2) =>
+      reduce({_ | _}, {_ |: _}, {_ | d2i(_)}, {d2i(_) |: _}, {_ | _}, res1, res2)
+    }
+
+    Valid(results)
+  }
+
+  val and = Directive { case (And(_), childResults) =>
+    val results: Result = childResults.reduce { (res1, res2) =>
+      reduce({_ & _}, {_ &: _}, {_ & d2i(_)}, {d2i(_) &: _}, {_ localAnd _}, res1, res2)
+    }
+
+    Valid(results)
+  }
+
+  val xor = Directive { case (Xor(_), childResults) =>
+    val rr: (RDD[(SpatialKey, Tile)], RDD[(SpatialKey, Tile)]) => RDD[(SpatialKey, Tile)] = { (r1, r2) =>
+      r1.combineValues(r2) { (t1, t2) => t1.combineDouble(t2)(gt.Xor.combine) }
+    }
+
+    val results: Result = childResults.reduce { (res1, res2) =>
+      reduce({_ ^ _}, {_ ^: _}, {_ ^ d2i(_)}, {d2i(_) ^: _}, rr, res1, res2)
+    }
+
+    Valid(results)
+  }
+
+  /* --- BINARY EXPRESSIONS --- */
+
+  val equalTo = Directive { case (Equal(_), res1 :: res2 :: Nil) =>
+    val results: Result =
+      reduce(
+        {_.localEqual(_)}, {(i,r) => r.localEqual(i)},
+        {_.localEqual(_)}, {(d,r) => r.localEqual(d)},
+        {_.localEqual(_)}, res1, res2)
+
+    Valid(results)
+  }
+
+  val unequalTo = Directive { case (Unequal(_), res1 :: res2 :: Nil) =>
+    val results: Result =
+      reduce(
+        {_.localUnequal(_)}, {(i,r) => r.localUnequal(i)},
+        {_.localUnequal(_)}, {(d,r) => r.localUnequal(d)},
+        {_.localUnequal(_)}, res1, res2)
+
+    Valid(results)
+  }
+
+  val lessThan = Directive { case (Less(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ < _}, {_ <<: _}, {_ < _}, {_ <<: _}, {_ < _}, res1, res2))
+  }
+
+  val lessThanOrEqualTo = Directive { case (LessOrEqual(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ <= _}, {_ <=: _}, {_ <= _}, {_ <=: _}, {_ <= _}, res1, res2))
+  }
+
+  val greaterThan = Directive { case (Greater(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ > _}, {_ >>: _}, {_ > _}, {_ >>: _}, {_ > _}, res1, res2))
+  }
+
+  val greaterThanOrEqualTo = Directive { case (GreaterOrEqual(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ >= _}, {_ >=: _}, {_ >= _}, {_ >=: _}, {_ >= _}, res1, res2))
+  }
+
+  // TODO The mask can probably be something other than `Polygon`.
+  val masking = Directive {
+    case (Masking(_), RDDResult(r) :: GeomResult(g: Polygon) :: Nil) => Valid(RDDResult(r.mask(g)))
+    case (Masking(_), GeomResult(g: Polygon) :: RDDResult(r) :: Nil) => Valid(RDDResult(r.mask(g)))
```

Comment:
`Polygon` probably not correct here.

Reply:
I returned an invalid if i couldn't `.as[MultiPolygon]` for the analogous TMS directive